pkg/sync/http/http_sync.go

```diff
@@ -82,6 +80,9 @@ func (hs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
 	})
 
 	hs.Cron.Start()
+
+	dataSync <- sync.DataSync{FlagData: fetch, Source: hs.URI, Type: sync.ALL}
+
```

Comment:
By moving this here, we know the add/start ran, which is what we assert in our tests, resolving the race.